go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go

```diff
@@ -707,6 +712,54 @@ func sendEventPoasInRange(event dataModel.Event) (error, int, string) {
 	return nil, -1, description
 }
 
+// sendEventScenarioUpdate - Process a scenario update event
+func sendEventScenarioUpdate(event dataModel.Event) (error, int, string) {
+	var err error
+	var description string
+
+	if event.EventScenarioUpdate == nil {
+		err := errors.New("Malformed request: missing EventScenarioUpdate")
+		return err, http.StatusBadRequest, ""
+	}
+	if len(event.EventScenarioUpdate.Nodes) == 0 {
+		err := errors.New("Malformed request: missing Nodes")
+		return err, http.StatusBadRequest, ""
+	}
+
+	// Perform necessary action on scenario
+	switch event.EventScenarioUpdate.Action {
+	case mod.ScenarioAdd:
+		description = "Added nodes ["
+		for i, node := range event.EventScenarioUpdate.Nodes {
+			if i != 0 {
+				description += ", "
+			}
+			description += node.Name
+		}
+		description += "]"
+		err = sbxCtrl.activeModel.AddScenarioNodes(&event.EventScenarioUpdate.Nodes)
+
+	case mod.ScenarioRemove:
+		description = "Removed nodes ["
+		for i, node := range event.EventScenarioUpdate.Nodes {
+			if i != 0 {
+				description += ", "
+			}
+			description += node.Name
+		}
+		description += "]"
+		err = sbxCtrl.activeModel.RemoveScenarioNodes(&event.EventScenarioUpdate.Nodes)
```

Comment:
I recommend removing node first, then based on error code, build the description.
e.g. Removed nodes[a,b,c] Failed to remove [d,e,f]

Reply:
Fixed in latest commit.
NOTE: Current implementation stops removing nodes if a node fails to be removed successfully.